src/types/Firestore.ts

```diff
@@ -1,88 +1,94 @@
-    venueEvents: Record<string, VenueEvent>;
-    userPurchaseHistory: Record<string, Purchase>;
-    partygoers: Record<string, User>;
-    users: Record<string, User>;
-    privatechats: Record<string, PrivateChatMessage>;
-    experiences: Record<string, Experience>;
-    eventPurchase: Record<string, Purchase>;
-    reactions: Record<string, Reaction>;
-    venues?: Record<string, AnyVenue>;
-    events?: Record<string, VenueEvent>;
-    playaVenues?: Record<string, AnyVenue>; // for the admin playa preview
-    allUsers?: Record<string, User>;
-    userModalVisits?: Record<string, UserVisit>;
-    userRoles: Record<string, Role>;
-    allowAllRoles: Record<string, Role>;
-    adminRole: AdminRole;
-  };
-  ordered: {
-    currentVenue: Array<WithId<AnyVenue>>;
-    currentEvent: Array<WithId<VenueEvent>>;
-    venueChats: Array<WithId<RestrictedChatMessage>>;
-    venueEvents: Array<WithId<VenueEvent>>;
-    userPurchaseHistory: Array<WithId<Purchase>>;
-    partygoers: Array<WithId<User>>;
-    users: Array<WithId<User>>;
-    privatechats: Array<WithId<PrivateChatMessage>>;
-    experiences: Array<WithId<Experience>>;
-    eventPurchase: Array<WithId<Purchase>>;
-    reactions: Array<WithId<Reaction>>;
-    venues?: Array<WithId<AnyVenue>>;
-    events?: Array<WithId<VenueEvent>>;
-    playaVenues?: Array<WithId<AnyVenue>>;
-    statsOnlineUsers?: Array<WithId<User>>;
-    statsOpenVenues?: Array<WithId<AnyVenue>>;
-    allUsers?: Array<WithId<User>>;
-    userModalVisits?: Array<WithId<UserVisit>>;
-    chatRequests?: Array<WithId<ChatRequest>>;
-  };
+  data: FirestoreData;
+  ordered: FirestoreOrdered;
+  status: FirestoreStatus;
+}
+
+export interface FirestoreStatus {
+  requesting: Record<ValidFirestoreKeys, boolean>;
+  requested: Record<ValidFirestoreKeys, boolean>;
+  timestamps: Record<ValidFirestoreKeys, number>;
+}
+
+// note: these entries should be sorted alphabetically
+export interface FirestoreData {
+  adminRole: AdminRole;
+  allowAllRoles: Record<string, Role>;
+  allUsers?: Record<string, User>;
+  currentEvent: Record<string, VenueEvent>;
+  currentVenue?: AnyVenue;
+  currentVenueEventsNG?: Record<string, VenueEvent>;
+  currentVenueNG?: AnyVenue;
+  eventPurchase: Record<string, Purchase>;
+  events?: Record<string, VenueEvent>;
+  experiences: Record<string, Experience>;
+  parentVenue?: AnyVenue;
+  partygoers: Record<string, User>;
+  playaVenues?: Record<string, AnyVenue>; // for the admin playa preview
+  privatechats: Record<string, PrivateChatMessage>;
+  reactions: Record<string, Reaction>;
+  userModalVisits?: Record<string, UserVisit>;
+  userPurchaseHistory: Record<string, Purchase>;
+  userRoles: Record<string, Role>;
+  users: Record<string, User>;
+  venueChats: Record<string, RestrictedChatMessage> | null;
+  venueEvents: Record<string, VenueEvent>;
+  venues?: Record<string, AnyVenue>;
+}
+
+// note: these entries should be sorted alphabetically
+export interface FirestoreOrdered {
```

Comment:
copying from original PR ([ref](https://github.com/sparkletown/sparkle/pull/721/files?file-filters%5B%5D=.ts&hide-deleted-files=true#r524930363))

Keeping these sorted alphabetically should make it easier when looking for keys